AIAgent/common/game.py

```diff
@@ -88,14 +89,9 @@ class MoveReward:
     ForCoverage: int
     ForVisitedInstructions: int
 
-    def __eq__(self, __o) -> bool:
-        if type(self) != type(__o):
-            raise RuntimeError(f"Can't compare {type(self)} with {type(__o)}")
-        return self == __o
-
     def __add__(self, __o: "MoveReward") -> "MoveReward":
-        if type(self) != type(__o):
-            raise RuntimeError(f"Can't add {type(__o)} to {type(self)}")
+        if not isinstance(__o, MoveReward):
+            raise NotImplementedError
```

Comment:
What is the reason to change exception?

Reply:
`NotImplementedError` lets other classes implement __add__, while `RuntimeError` just halts execution

Reply:
Is it a good idea on our case? @emnigma ? In my opinion, it is unlikely reasonable to implement other class to ovewrite add. But it is important to get detailed information about the problem.

Reply:
What's wrong with NIE? It also provides detailed information by clearly signaling an unimplemented method.

Reply:
I think you're right. Does `TypeError` seem to be more appropriate here?